internal/js/modules/k6/browser/browser/route_mapping.go

```diff
@@ -0,0 +1,62 @@
+package browser
+
+import (
+	"context"
+
+	"github.com/grafana/sobek"
+
+	"go.k6.io/k6/internal/js/modules/k6/browser/common"
+	"go.k6.io/k6/internal/js/modules/k6/browser/k6ext"
+)
+
+// mapRoute to the JS module.
+func mapRoute(vu moduleVU, route *common.Route) mapping {
+	return mapping{
+		"abort": func(reason string) *sobek.Promise {
+			return k6ext.Promise(vu.Context(), func() (any, error) {
+				return nil, route.Abort(reason)
+			})
+		},
+		"fulfill": func(opts sobek.Value) *sobek.Promise {
+			fopts := parseFulfillOptions(vu.Context(), opts)
+			return k6ext.Promise(vu.Context(), func() (any, error) {
+				return nil, route.Fulfill(fopts)
+			})
+		},
+		"request": func() mapping {
+			return mapRequest(vu, route.Request())
+		},
+	}
+}
+
+func parseFulfillOptions(ctx context.Context, opts sobek.Value) common.FulfillOptions {
+	fopts := common.FulfillOptions{}
+	if !sobekValueExists(opts) {
+		return fopts
+	}
+
+	rt := k6ext.Runtime(ctx)
+	obj := opts.ToObject(rt)
+	for _, k := range obj.Keys() {
+		switch k {
+		case "body":
+			fopts.Body = obj.Get(k).String()
+		case "contentType":
+			fopts.ContentType = obj.Get(k).String()
+		case "headers":
+			headers := obj.Get(k).ToObject(rt)
+			headersKeys := headers.Keys()
+			fopts.Headers = make([]common.HTTPHeader, len(headersKeys))
+			for i, hk := range headersKeys {
+				fopts.Headers[i] = common.HTTPHeader{
+					Name:  hk,
+					Value: headers.Get(hk).String(),
+				}
+			}
+		case "status":
+			fopts.Status = obj.Get(k).ToInteger()
+		}
```

Comment:
Looking at https://playwright.dev/docs/api/class-route#route-fulfill - you are not handling `path`, `json` and `response` . Can we have a comment here. Also I feel like handling all but `response` is likely fairly quick :thinking: .

But if not maybe lets throw an exception instead of silently doing the wrong thing

Reply:
Updated in https://github.com/grafana/k6/pull/4961/commits/67f24be3c8cb74ac463af2cf62248fecba3024e0